src/arch/amdgpu/vega/tlb_coalescer.cc

```diff
@@ -160,12 +160,42 @@ VegaTLBCoalescer::updatePhysAddresses(PacketPtr pkt)
     Addr first_entry_vaddr = tlb_entry.vaddr;
     Addr first_entry_paddr = tlb_entry.paddr;
     int page_size = tlb_entry.size();
+
+    observedPagesize.insert(page_size);
+
+    Addr virt_page_addr;
+
+    // Find coalesced translation request.
+    for (auto pgsize_seen : observedPagesize) {
+        virt_page_addr = roundDown(pkt->req->getVaddr(), pgsize_seen);
+        if (issuedTranslationsTable.count(virt_page_addr) != 0)
+            break;
+    }
+
+    DPRINTF(GPUTLB, "Update phys. addr. for %d \
+            coalesced reqs for page %#x\n",
+            issuedTranslationsTable[virt_page_addr].size(),
+            virt_page_addr);
+
+
     bool uncacheable = tlb_entry.uncacheable();
     int first_hit_level = sender_state->hitLevel;
     bool is_system = pkt->req->systemReq();
 
-    for (int i = 0; i < issuedTranslationsTable[virt_page_addr].size(); ++i) {
+    for (int i = 0;
+         i < issuedTranslationsTable[virt_page_addr].size(); ++i) {
         PacketPtr local_pkt = issuedTranslationsTable[virt_page_addr][i];
+
+        Addr local_pkt_vaddr = local_pkt->req->getVaddr();
+
+        //check if the local packet vaddr belongs to the page translated,
```

Comment:
I think "page translated" is a bit confusing here.  Maybe something like "check if the returned page translation's virtual address matches any of the pending requests.  If not, reissue pending request as a 4K page" would make more sense?